common/workunit/workunit.cpp

```diff
@@ -11216,6 +11217,16 @@ __int64 CLocalWUResult::getResultTotalRowCount() const
     return p->getPropInt64("totalRowCount", -1);
 }
 
+StringBuffer &CLocalWUResult::getResultTotalRowCountString(StringBuffer &str) const
```

Comment:
discussion/observation.  I am not sure this is the cleanest way to implement this functionality for a couple of reasons.

i) It is moving esp-specific code into the general purpose workunit code.  It is very specific in what the new function does and I doubt anything else is going to be calling this function.

ii) It is complicating the public interface of a class by adding a virtual function that can already be implemented in terms of the other public functions.

It does have the advantage of hiding the fact that -1 means that the result count is not known.

An alterive would be a static function:
```
StringBuffer & getResultTotalRowCountString(IConstWUResult & result, IStringBuffer &str) const
{
    __int64 count = result. getResultTotalRowCount();
    if (count == -1)
        str.append("[??? rows]");
    else
        str.append('[').append(count).append(" rows]");
    return str;
}
```
which would live in an esp library if shared functions.

You could introduce a hasTotalRowCount() function to IConstWUResult to avoid the need for checking for -1.